processing/src/main/java/org/apache/druid/frame/read/columnar/StringFrameColumnReader.java

```diff
@@ -92,17 +92,18 @@ public Column readRACColumn(Frame frame)
     final Memory memory = frame.region(columnNumber);
     validate(memory);
 
-    final long positionOfLengths = getStartOfStringLengthSection(frame.numRows(), false);
-    final long positionOfPayloads = getStartOfStringDataSection(memory, frame.numRows(), false);
+    final boolean multiValue = isMultiValue(memory);
+    final long positionOfLengths = getStartOfStringLengthSection(frame.numRows(), multiValue);
+    final long positionOfPayloads = getStartOfStringDataSection(memory, frame.numRows(), multiValue);
 
     StringFrameColumn frameCol =
         new StringFrameColumn(
             frame,
-            false,
+            multiValue,
             memory,
             positionOfLengths,
             positionOfPayloads,
-            asArray || isMultiValue(memory) // Read MVDs as String arrays
+            asArray || multiValue // Read MVDs as String arrays
```

Comment:
Is the comment still relevant, or needs an update?

Reply:
yes. it is relevant (if its an actual MVD column, we still read it as array)